src/dynamo-streams.test.ts

```diff
@@ -508,4 +514,227 @@ describe('DynamoStreamHandler', () => {
+      })
+        .onInsert(async (ctx, item) => {
+          mocks.insert.started(Date.now(), item);
+          await wait(100);
+          mocks.insert.finished(Date.now(), item);
+        })
+        .onModify(async (ctx, oldItem, newItem) => {
+          mocks.modify.started(Date.now(), oldItem, newItem);
+          await wait(100);
+          mocks.modify.finished(Date.now(), oldItem, newItem);
+        })
+        .onRemove(async (ctx, item) => {
+          mocks.remove.started(Date.now(), item);
+          await wait(100);
+          mocks.remove.finished(Date.now(), item);
+        })
+        .lambda();
+
+      const start = Date.now();
+
+      await handler(
+        {
+          Records: [
+            {
+              eventName: 'INSERT',
+              dynamodb: {
+                Keys: marshall({ id: 'test-id-1', name: 'test-name-1' }) as any,
+                NewImage: marshall({
+                  id: 'test-id-1',
+                  name: 'test-name-1',
+                  otherValue: 'test-value-1',
+                }) as any,
+              },
+            },
+            {
+              eventName: 'MODIFY',
+              dynamodb: {
+                Keys: marshall({ id: 'test-id-1', name: 'test-name-1' }) as any,
+                NewImage: marshall({
+                  id: 'test-id-1',
+                  name: 'test-name-1',
+                  otherValue: 'test-value-1',
+                }) as any,
+                OldImage: marshall({
+                  id: 'test-id-1',
+                  name: 'test-name-1',
+                  otherValue: 'test-value-2',
+                }) as any,
+              },
+            },
+            {
+              eventName: 'MODIFY',
+              dynamodb: {
+                Keys: marshall({ id: 'test-id-2', name: 'test-name-1' }) as any,
+                NewImage: marshall({
+                  id: 'test-id-2',
+                  name: 'test-name-1',
+                  otherValue: 'test-value-1',
+                }) as any,
+                OldImage: marshall({
+                  id: 'test-id-2',
+                  name: 'test-name-1',
+                  otherValue: 'test-value-2',
+                }) as any,
+              },
+            },
+            {
+              eventName: 'REMOVE',
+              dynamodb: {
+                Keys: marshall({ id: 'test-id-1', name: 'test-name-1' }) as any,
+                OldImage: marshall({
+                  id: 'test-id-1',
+                  name: 'test-name-1',
+                  otherValue: 'test-value-2',
+                }) as any,
+              },
+            },
+          ],
+        },
+        {} as any,
+        null as any,
+      );
+
+      const end = Date.now();
+
+      // This assertion confirms that the group doesn't process in less than 200ms.
+      // If it did, then some of the ordered events would be parallelized, which would be bad.
+      expect(end - start).toBeGreaterThanOrEqual(300);
```

Comment:
I think the comment above is outdated and should say `300ms`.

Reply:
yes

Reply:
I think it might be a bit safer, and more certain, if we refactored this with jest mocked timers?